### PR TITLE
Update document, dependency-specification.md

### PR DESCRIPTION
**Pull request by squirrel532**
Add description of `source` property. Enhance doc of #908 .

I search around and find this feature in issue, it will be good if we write it down to documents.

# Pull Request Check List

Resolves:  -

- [ ] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by squirrel532** [comment]
push up

File: docs/docs/dependency-specification.md [hunk 1]
```diff
@@ -139,6 +139,20 @@ with the corresponding `add` call:
 poetry add https://example.com/my-package-0.1.0.tar.gz
 ```
 
+## `source` dependencies
+To dependent on a package located in the private repositry `my-private-repo`,
+you can use `source` property:
```
Comment (abn) [inline docs/docs/dependency-specification.md:144]:
```suggestion
To depend on a package from an [alternate repository](/docs/repositories/#install-dependencies-from-a-private-repository)
you can use `source` property:
```

File: docs/docs/dependency-specification.md [hunk 2]
```diff
@@ -139,6 +139,20 @@ with the corresponding `add` call:
...
+```toml
+[[tool.poetry.source]]
+name = "my-private-repo"
+url = "https://example.com/"
+secondary = true
+
+[tool.poetry.dependencies]
+my-cool-package = { version = "*", source = "my-private-repo" }
+```
```
Comment (abn) [inline docs/docs/dependency-specification.md:153]:
```suggestion
```toml
[[tool.poetry.source]]
name = "foo"
url = "https://foo.bar/simple/"
secondary = true

[tool.poetry.dependencies]
my-cool-package = { version = "*", source = "foo" }
```
```

Let's keep the example consistent.

File: docs/docs/dependency-specification.md [hunk 3]
```diff
@@ -139,6 +139,20 @@ with the corresponding `add` call:
...
+By setting `secondary = true`, poetry will still pull other dependencies from Pypi.
+See [Using a private repository](repositories.md#using-a-private-repository) for futher information.
```
Comment (abn) [inline docs/docs/dependency-specification.md:155]:
```suggestion
!!!note
    In this example, we expect `foo` to be configured conrrectly. See [using a private repository](repositories.md#using-a-private-repository) for futher information.
```

**Review by abn — CHANGES_REQUESTED** [review]

**Review by stephsamson — APPROVED** [review]

**Comment by squirrel532** [comment]
push again. :)

File: docs/docs/dependency-specification.md [hunk 4]
```diff
@@ -139,6 +139,19 @@ with the corresponding `add` call:
 poetry add https://example.com/my-package-0.1.0.tar.gz
 ```
 
+## `source` dependencies
+To depend on a package from an [alternate repository](/docs/repositories/#install-dependencies-from-a-private-repository)
```
Comment (ggabriel96) [inline docs/docs/dependency-specification.md:143]: Just suggesting a newline between the heading and the content.

```suggestion

To depend on a package from an [alternate repository](/docs/repositories/#install-dependencies-from-a-private-repository),
```

File: docs/docs/dependency-specification.md [hunk 5]
```diff
@@ -139,6 +139,19 @@ with the corresponding `add` call:
...
+you can use `source` property:
```
Comment (ggabriel96) [inline docs/docs/dependency-specification.md:144]: Minor thing and a newline

```suggestion
you can use the `source` property:

```

File: docs/docs/dependency-specification.md [hunk 6]
```diff
@@ -139,6 +139,19 @@ with the corresponding `add` call:
...
+```toml
+[[tool.poetry.source]]
+name = "foo"
+url = "https://foo.bar/simple/"
+secondary = true
+
+[tool.poetry.dependencies]
+my-cool-package = { version = "*", source = "foo" }
```
Comment (ggabriel96) [inline docs/docs/dependency-specification.md:152]: You forgot to close the code block here (also added a newline).

````suggestion
my-cool-package = { version = "*", source = "foo" }
```

````

**Review by ggabriel96 — COMMENTED** [review]
I would like to suggest a few changes, if I may.
This is how the doc is currently being rendered:

image

File: docs/docs/dependency-specification.md [hunk 7]
```diff
@@ -139,6 +139,23 @@ with the corresponding `add` call:
...
+```
+
+!!!note
+    In this example, we expect `foo` to be configured conrrectly. See [using a private repository](repositories.md#using-a-private-repository) for futher information.
```
Comment (ggabriel96) [inline docs/docs/dependency-specification.md:158]: Found a typo XD (there was an `n` in `correctly`)

Also added a newline to be similar to the other notes here in this document.

```suggestion
!!!note

    In this example, we expect `foo` to be configured correctly. See [using a private repository](repositories.md#using-a-private-repository) for futher information.
```

Comment (squirrel532) [reply docs/docs/dependency-specification.md:158]: it's really hard to find that. XD

File: docs/docs/dependency-specification.md [hunk 8]
```diff
@@ -139,6 +139,24 @@ with the corresponding `add` call:
...
+url = "https://foo.bar/simple/"
+secondary = true
+
+[tool.poetry.dependencies]
+my-cool-package = { version = "*", source = "foo" }
+```
+
+!!!note
+
+    In this example, we expect `foo` to be configured correctly. See [using a private repository](repositories.md#using-a-private-repository) for futher information.
```
Comment (ggabriel96) [inline docs/docs/dependency-specification.md:159]: Sorry, I missed this one :sweat_smile: 

```suggestion
    In this example, we expect `foo` to be configured correctly. See [using a private repository](repositories.md#using-a-private-repository) for further information.
```

File: docs/docs/dependency-specification.md [hunk 9]
```diff
@@ -139,6 +139,24 @@ with the corresponding `add` call:
...
+```toml
+[[tool.poetry.source]]
+name = "foo"
+url = "https://foo.bar/simple/"
+secondary = true
+
+[tool.poetry.dependencies]
+my-cool-package = { version = "*", source = "foo" }
+```
+
```
Comment (ggabriel96) [inline docs/docs/dependency-specification.md:156]: By the way, I just noticed this section doesn't mention the `add` command. So, similar to the `url` section just above this one, I would add something like:

````suggestion

with the corresponding `add` call:

```sh
poetry add my-cool-package --source foo
```

````

**Review by neersighted — CHANGES_REQUESTED** [review]
This overlaps with #3477, but still adds some clarification/information on top that is valuable. A rebase on top of the changes to master with a focus on being clear and concise would be appreciated!

**Review by neersighted — APPROVED** [review]

**Comment by neersighted** [comment]
It looks like this isn't actually on the latest `master`. The required checks changed a bit -- can you fetch `origin/master`, rebase again, and then ping me?

**Comment by squirrel532** [comment]
@neersighted  Not sure what happened to my local repo, but it's ok now.

**Review by neersighted — APPROVED** [review]

**Comment by squirrel532** [comment]
@abn could you help review again ?

**Pull request merged**